src/components/TempSplitter.tsx

```diff
@@ -151,6 +151,10 @@ export const TempSplitter = ({ firName }: { firName: string }) => {
       firName: firName,
     });
 
+  // TODO: We have already retrieved the latest facility data in MainMap.tsx
+  //       Do not make any new requests here, just figure out how to share
+  //       the data from the MainMap to the TempSplitter. Is there somewhere
```

Comment:
TempSplitter has been entirely removed on the splitconfig branch in place of the actual page, this logic was just temporary for sake of the ZDC demo. 

As to the larger point of using context or something else to pass down data, I don't actually think we need to use context. The MainMap should reflect the latest data, and continue to refresh and fetch if it has outdated information. As for the splitconfig page, we *will* need to make another request on every fresh render of the splitconfig page because we want to show what the latest group data is whenever I go in and begin to edit a split. I won't however want that page to constantly refresh as I am regrouping sectors, which would be quite frustrating as a user. This means meaning that we actually don't want to reference a master context with constantly-updating data. If this is the case, there is no other component besides MainMap where this data would be relevant, and in the interest of not over-utilizing context, we can probably isolate the fetch and refresh logic to that component. If there are other components that require this constantly up-to-date data, then context might be appropriate.

This data would only change, by the way, when controllers are simultaneously attempting to edit the groupings of the same facility. The consequences of this are that changes made by the latest-submitting user will override the grouping of the user who submitted first, but I think that is acceptable and expected behavior if you are editing the groupings of the same facility.

Reply:
The solution to the problem you proposed is not to provide constant real-time facilityData and groupData updates, but to have an external system (a different system) for tracking and detecting whether another user is on the page making edits or not. 

Assuming we did provide the constant updates for facility/group data, there are two big issues:
1) The UI shifting around in the middle of a configuration session is terrible for UX because it's confusing, frustrating, and would lead to mistakes being made. Consider the case where a group you are in the middle of editing suddenly disappears because another user deleted it. Consider another case where you don't notice that sector *X* was automatically and quietly moved from group "B" to "C" because you were focusing on group "E" at the time. Maybe you go back and are confused when you notice the inconsistency, maybe you don't catch it. Either way is a very bad thing for UX's sake. The only time where this type of behavior is reasonable is when it comes to real-time collaboration and you can explicitly see who is making changes and to what, which we don't support.

2) Even if we provided the same type of data that the MainMap receives to the splitconfig page, it still doesn't prevent the case where two users don't realize they are editing the same thing until its too late. Only when one person is done and submits would the other user realize that edits have been made, at which point both users are not only deep into configuration, but also means that the first user might not check back after submitting, and have their changes overwritten anyway. 

In summary, I don't think we need to expose the constantly changing group/facility data, though we might need a different system for handling editing conflicts.

Reply:
I don't really know what to say other than... I totally disagree.

Having "an external system for tracking and detecting whether another user is on the page making edits" may be a handy addition that could be added post MVP. Even then, I would say this would objectively be of very low value. But for the time being, we don't need any of it.

You're sort of blowing the "problem" out of proportion and prioritizing the wrong things. Cases where multiple people are simultaneously editing a split will always be extremely rare. And you are pushing for a system that gives priority for handling this rare scenario, at the expense of behavior in the 99.9% of cases where only one person is editing. Every single thing you mention in point 1 is somewhere between acceptable behavior and actually_desirable behavior_. Not one of those behaviors is even remotely a negative. I understand you have experience in software and UX that says "this is not how this should work", and you perceive the problem as being significant. But I need to have some degree of latitude to say what is and is not important for the people who would use the tool, as I am the one with the experience of what is important to the center controllers, TMUs, Command Center, etc, and how the tool should behave for them. To that end, I am saying that the protections you're trying to establish for simultaneous edits _are not remotely important_-- the things you list as problems are not problems at all, whereas protecting the simple concept of "anything you ever see anywhere is always completely up to date at all times for everyone" is critically important. We should not degrade the behavior under normal conditions for an edge-case scenario which has no downsides.